fix: beaconcha.in API rate limiting

wackerow opened the pull request:
## Description
- Add proxy API call to beaconchain API, using key via Netlify serverless function
- Falls back to direct key-less API query for development mode
- 1 second "waiting" timeout in between calls that immediately proceed each other to obey rate limits

## Preview link
https://deploy-preview-794--phenomenal-frangipane-7c4bd1.netlify.app/

## Related issue
- Fixes #791

wackerow committed:
fix: beaconcha.in API rate limiting

Add proxy API call with key; Falls back to direct key-less API query for development mode

valefar-on-discord reviewed (approved):
I didn't test the netlify side but it everything looks correct.

valefar-on-discord committed:
Add 1 second delay after withdrawal address api request

valefar-on-discord commented:
Fixes #791

wackerow committed:
patch: access to BEACONCHAIN_URL env var

pass BEACONCHAIN_URL env var to serverless fnc via getBeaconchainV1ApiUrl helper function

valefar-on-discord reviewed (approved):
Tested on preview with hoodi and works
